packages/manager/src/features/Events/EventsLanding.tsx

```diff
@@ -29,8 +30,16 @@ interface Props {
 
 export const EventsLanding = (props: Props) => {
   const { emptyMessage, entityId } = props;
+  const { isTaxIdEnabled } = useIsTaxIdEnabled();
 
-  const filter: Filter = { action: { '+neq': 'profile_update' } };
+  const filter: Filter = {
+    action: {
+      '+and': [
+        { '+neq': 'profile_update' },
+        isTaxIdEnabled ? { '+neq': 'tax_id_invalid' } : {},
+      ],
+    },
```

Comment:
This filter isn't valid

![Screenshot 2024-06-05 at 11 05 52 AM](https://github.com/linode/manager/assets/115251059/47bd767f-2014-4f34-a2fc-dcfe61879f43)

This should do what you are intending:
```suggestion
'+and': [
      { action: { '+neq': 'profile_update' } },
      ...(isTaxIdEnabled ? [{ action: { '+neq': 'tax_id_invalid' } }] : []),
    ],
```

I'm fine with this change, but do feel like it might be an over-flag-ization. I feel like the API should be responsible for hiding/showing **and/or** I wonder if we even need to hide it because it will only start showing up for a customer once it has been enabled on the API-side

Also, this goes back to what @abailly-akamai and I were talking about [here](https://linode.slack.com/archives/GUN7PKZR6/p1717519268385269?thread_ts=1717517920.519189&cid=GUN7PKZR6). Ideally, this API filter should eventually be shared with the Event Menu, but no action is needed now. @abailly-akamai or I will address that later

Reply:
Correct - this is about to be handled and we will be able to configure the filter so that it's sharable for EventsLanding and the NotificationCenter. I would suggest removing this change and add a TODO

Reply:
Sounds good - I'll revert if that's the case